### PR TITLE
Add architecture docs

### PR DESCRIPTION
**Pull request by devongovett**
This adds architecture docs, and moves the pages on the homepage into categories.

### CODE REVIEW

File: packages/dev/docs/pages/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+the ability to reuse behavior between multiple components. Hooks allow
+accessing React features like state and effects from functions which
+can be called from any component. If you're unfamiliar with hooks,
+we recommend reading the documentation linked above first.
+
+### State hook
+
+At the bottom is the state hook. This hook is shared across platforms —
+it could work on the web, react-native, or any other platform, and makes
+not assumption about the view system it is running on. It also has no
```
Comment (snowystinger) [inline packages/dev/docs/pages/architecture.mdx:101]: not -> no
assumption -> assumptions

File: packages/dev/docs/pages/architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+### State hook
+
+At the bottom is the state hook. This hook is shared across platforms —
+it could work on the web, react-native, or any other platform, and makes
+not assumption about the view system it is running on. It also has no 
+theme or design system specific logic.
+
+The state hook accepts common props from the component and provides 
+state management. It returns an interface for reading and updating
+component state, with methods that implement much of the core logic
```
Comment (snowystinger) [inline packages/dev/docs/pages/architecture.mdx:106]: i think this comma is extraneous? the sentence feels awkward though, I can't quite tell if you're saying with as in, in addition to, or if you're saying the updating of the component state is done with...
Maybe, "It implements the core logic for the component and returns an interface for reading and updating the component state." ?

File: packages/dev/docs/pages/architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+Not all components have a state hook. For example, many components are
+read-only — they display something to the user but don't allow them
+to change it. State hooks are found in interactive components that allow
+data entry, or have some kind of visual state that the user can update
+(e.g. expanding/collapsing).
+
+### Behavior hook
+
+In the middle is the behavior hook. This hook is platform specific, and
+depends on the platform API (e.g. the DOM or react-native). It still has
```
Comment (snowystinger) [inline packages/dev/docs/pages/architecture.mdx:118]: It also has, instead of still

File: packages/dev/docs/pages/architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+At the top is the component, which lives in each design system and composes 
+all of these pieces together. It provides the theme and design system specific
+logic, and renders the actual platform elements. It applies styles, which could
+be implemented in many different ways (e.g. CSS classes, CSS-in-JS, etc.).
+
+The component uses props returned by the behavior hook and state from the state 
+hook to implement the visual appearance. It spreads props returned from the behavior
+hook onto elements that it renders to apply semantics and interactions, and can use
+the state from the state hook to adjust its visual appearance based on the state.
```
Comment (snowystinger) [inline packages/dev/docs/pages/architecture.mdx:146]: redundant to say 'based on the state'

File: packages/dev/docs/pages/architecture.mdx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+component logic between design systems and across platforms. For example,
+user interactions, accessibility, internationalization, and behavior
+can be reused, while allowing custom styling and rendering to live
+within individual design systems. This has the potential to improve
+the overall quality of applications, while saving companies money
+and time, and reducing duplicated effort across the industry.
+
+## Architecture
+
+<figure className={style.float}>
```
Comment (dannify) [inline packages/dev/docs/pages/architecture.mdx:61]: 😍

File: packages/dev/docs/pages/architecture.mdx [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+Within a component, the hooks are designed to be **highly composable**,
+with individual features split into many hooks. This allows you to combine 
+them together and achieve your design with only the features you need in 
+your design system.
+
+The overall goal for the project is to make reusing behavior across design
+systems as easy as possible, while allowing full design customizability
+and avoiding code bloat.
+
+<div style={{clear: 'both'}} />
```
Comment (dannify) [inline packages/dev/docs/pages/architecture.mdx:188]: image

**Pull request merged**